source/braille.py

```diff
@@ -3841,3 +3845,18 @@ def _speakOnRouting(info: textInfos.TextInfo):
 
 	info.expand(textInfos.UNIT_CHARACTER)
 	spellTextInfo(info)
+
+
+def _speakOnNavigatingByUnit(info: textInfos.TextInfo):
+	"""Speaks the reading unit after navigating by it with braille.
+
+	:param info: The reading unit TextInfo at the cursor position after navigating.
+	"""
+	if not config.conf["braille"]["speakOnNavigatingByUnit"]:
+		return
+	# Import late to avoid circular import.
+	from speech.speech import cancelSpeech, speakTextInfo
+
+	readingUnit = handler.buffer.regions[-1]._getReadingUnit()
```

Comment:
Rather than getting the actual unit this way, I'd add it as a parameter to `_speakOnNavigatingByUnit` instead.